packages/hop-node/src/db/TransferRootsDb.ts

```diff
@@ -133,34 +135,20 @@ class TransferRootsDb extends BaseDb {
     })
   }
 
-  // TODO: This should be a new DB for a TransferBond, not a TransferRoot
-  // This will add new requirements to this return statement
   async getChallengeableTransferRoots (
     filter: Partial<TransferRoot> = {}
   ): Promise<TransferRoot[]> {
     const transfers = await this.getTransferRoots()
     return transfers.filter(item => {
+      // Do not check if a rootHash has been committed. A rootHash can be committed and bonded,
+      // but if the bond uses a different totalAmount then it is fraudulent. Instead, use the
+      // transferRootId. If transferRootIds do not match then we know the bond is fraudulent.
+      const isTransferRootIdValid = item.bondTransferRootId === item.transferRootId
```

Comment:
Maybe I missed it but where does `bondTransferRootId` get set before any challenges?

Reply:
You are correct.

`bondTransferRootId` (and `bondedAt`) were incorrectly placed in the `handleTransferBondChallengedEvent()` instead of `handleTransferRootBondedEvent()`. 

They will be moved.

Reply:
Updated 639073aa3c32acdaec0ecb6407c418ef073c8d58 and d8db75760d96b7e9754dac0c8f049a11da073376

Reply:
perfect

Reply:
Just to confirm, isTransferRootIdValid will be false if only the transfer root is bonded and there's no commit event correct (because the transferRootId is not set elsewhere)?

Reply:
That is correct.